### PR TITLE
[RDBMS] create params as per new business model

### PR DESCRIPTION
**Pull request by praveenbarli**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -24,26 +24,29 @@ def _complex_params(command_group, engine):
 
         with self.argument_context('{} server create'.format(command_group)) as c:
             c.expand('sku', engine.models.Sku)
-            c.ignore('name', 'family', 'size')
+            c.ignore('size')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: It is intended that these parameters are changed both for postgres and MySQL?

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 2]
```diff
@@ -65,8 +68,8 @@ def _complex_params(command_group, engine):
             c.argument('administrator_login', options_list=['--admin-user', '-u'])
             c.argument('administrator_login_password', options_list=['--admin-password', '-p'], required=False, help='The password of the administrator login.')
             c.argument('ssl_enforcement', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--ssl-enforcement'], help='Enable ssl enforcement or not when connect to server.')
-            c.argument('tier', arg_type=get_enum_type(['Basic', 'Standard']), options_list=['--performance-tier'], help='The performance tier of the server.')
-            c.argument('capacity', options_list=['--compute-units'], type=int, help='Number of compute units.')
+            c.argument('tier', arg_type=get_enum_type(['Basic', 'General Purpose', 'Standard']), options_list=['--performance-tier'], help='The performance tier of the server.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Test out "General Purpose"... argparse may not like that.  You might need to call it "GeneralPurpose" and convert it code if your service needs the string "General Purpose".

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 3]
```diff
@@ -65,8 +68,8 @@ def _complex_params(command_group, engine):
...
+            c.argument('capacity', options_list=['--vCore'], type=int, help='Number of vCore.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: should be `--vcore`.  We don't have any options in the CLI that are uppercase.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/validators.py [hunk 4]
```diff
@@ -24,11 +24,6 @@ def _final_validator_impl(cmd, namespace):
         for validator in validators:
             validator(namespace)
 
-        if namespace.sku.tier or namespace.sku.capacity:
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/validators.py]: Why are these being removed?

**Comment by tjprescott** [comment]
Are these changes breaking?  Also, you need to bump the version of your module for these changes.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/validators.py (hunk 4 above)
Comment (praveenbarli) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/validators.py]: Skuname wasn't a required field earlier, for creating a server; But, now it is. So, this validation is no longer applicable.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (praveenbarli) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: yes

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (praveenbarli) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Actually, it should be GeneralPurpose. Changed it

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py [hunk 5]
```diff
@@ -213,14 +216,6 @@ def test_mysql_proxy_resources_mgmt(self, resource_group, server, database_engin
...
         self._test_log_file_mgmt(resource_group, server, database_engine)
 
-    @ResourceGroupPreparer()
-    @ServerPreparer(engine_type='postgres')
-    def test_postgres_proxy_resources_mgmt(self, resource_group, server, database_engine):
-        self._test_firewall_mgmt(resource_group, server, database_engine)
-        self._test_db_mgmt(resource_group, server, database_engine)
-        self._test_configuration_mgmt(resource_group, server, database_engine)
-        self._test_log_file_mgmt(resource_group, server, database_engine)
-
```
Comment (akamath46297) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py]: Please add these tests back after validate Postgres

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py [hunk 6]
```diff
@@ -37,33 +43,56 @@ def _server_restore(cmd, client, resource_group_name, server_name, parameters, *
...
+                                    storage_profile=instance.storage_profile,
                                     administrator_login_password=administrator_login_password,
                                     version=None,
                                     ssl_enforcement=ssl_enforcement,
                                     tags=tags)
 
     return params
 
+def _create_server_for_mysql(cmd, client, resource_group_name, server_name, parameters, **kwargs):
+     return client.create_or_update(resource_group_name, server_name, parameters)
```
Comment (Suna-MS) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py]: I prefer to update the swagger file to keep the operationId of PUT server the same for mysql and pg, then you don't have to define two function for pg & mysql. (make them both client.create())

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py [hunk 7]
```diff
@@ -37,33 +43,56 @@ def _server_restore(cmd, client, resource_group_name, server_name, parameters, *
     except Exception as e:
         raise ValueError('Unable to get source server: {}.'.format(str(e)))
 
+def _restore_mysql(cmd, client, resource_group_name, server_name, parameters, **kwargs):
+    _server_restore(cmd, client, resource_group_name, server_name, parameters, **kwargs)
```
Comment (Suna-MS) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py]: Same for _create_server_for_xxx, please work with qingqing to update the swagger file, and make the create server operation id the same for mysql and pg

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py [hunk 8]
```diff
@@ -39,15 +40,18 @@ def __init__(self, engine_type='mysql', engine_parameter_name='database_engine',
         self.admin_password = admin_password
         self.resource_group_parameter_name = resource_group_parameter_name
         self.skip_delete = skip_delete
+        self.sku_name = sku_name
+        
 
     def create_resource(self, name, **kwargs):
         group = self._get_resource_group(**kwargs)
-        template = 'az {} server create -l {} -g {} -n {} -u {} -p {}'
+        template = 'az {} server create -l {} -g {} -n {} -u {} -p {} --sku-name {}'
```
Comment (Suna-MS) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py]: We should support create server with only required parameters (location/resourcegroup/name/adminUser/adminPassword), sku is optional.  Please remove --sku-name here.

**Comment by tjprescott** [comment]
@praveenbarli are these still changes you want in the CLI?  If so, please rebase.

**Pull request closed without merging**